aci/resource_aci_cloudtemplateregiondetail.go

```diff
@@ -0,0 +1,183 @@
+)
+
+func resourceAciCloudTemplateRegion() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAciCloudTemplateRegionCreate,
+		UpdateContext: resourceAciCloudTemplateRegionUpdate,
+		ReadContext:   resourceAciCloudTemplateRegionRead,
+		DeleteContext: resourceAciCloudTemplateRegionDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceAciCloudTemplateRegionImport,
+		},
+
+		SchemaVersion: 1,
+		Schema: AppendBaseAttrSchema(AppendNameAliasAttrSchema(map[string]*schema.Schema{
+			"parent_dn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"hub_networking": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"disabled",
+					"enabled",
+				}, false),
+			},
+		})),
+	}
+}
+
+func getRemoteCloudTemplateRegion(client *client.Client, dn string) (*models.CloudTemplateRegion, error) {
+	cloudTemplateRegionDetailCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	cloudTemplateRegionDetail := models.CloudTemplateRegionFromContainer(cloudTemplateRegionDetailCont)
+	if cloudTemplateRegionDetail.DistinguishedName == "" {
+		return nil, fmt.Errorf("Cloud Template Region %s not found", dn)
+	}
+	return cloudTemplateRegionDetail, nil
+}
+
+func setCloudTemplateRegionAttributes(cloudTemplateRegionDetail *models.CloudTemplateRegion, d *schema.ResourceData) (*schema.ResourceData, error) {
+	d.SetId(cloudTemplateRegionDetail.DistinguishedName)
+	cloudTemplateRegionDetailMap, err := cloudTemplateRegionDetail.ToMap()
+	if err != nil {
+		return d, err
+	}
+	dn := d.Id()
+	if dn != cloudTemplateRegionDetail.DistinguishedName {
+		d.Set("parent_dn", "")
+	} else {
+		d.Set("parent_dn", GetParentDn(cloudTemplateRegionDetail.DistinguishedName, fmt.Sprintf("/"+models.RnCloudtemplateRegionDetail)))
+	}
+	d.Set("annotation", cloudTemplateRegionDetailMap["annotation"])
+	d.Set("hub_networking", toggleOptions(cloudTemplateRegionDetailMap["hubNetworkingEnabled"]))
+	return d, nil
+}
+
+func resourceAciCloudTemplateRegionImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	cloudTemplateRegionDetail, err := getRemoteCloudTemplateRegion(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setCloudTemplateRegionAttributes(cloudTemplateRegionDetail, d)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciCloudTemplateRegionCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] CloudTemplateRegion: Beginning Creation")
+	aciClient := m.(*client.Client)
+	CloudProviderandRegionNamesDn := d.Get("parent_dn").(string)
```

Comment:
Reason to have capitalised variable?
```suggestion
	cloudProviderandRegionNamesDn := d.Get("parent_dn").(string)
```

Reply:
that applies for more variable assignments in your PR, like below also annotation